### PR TITLE
Fix inputs to octokit/request-action workflow call

### PR DESCRIPTION
**Pull request by addyess**
The octokit/request-action@v2.3.0 doesn't include these arguments `repo`, `owner` and `branch`.  

https://github.com/octokit/request-action/blob/v2.3.0/action.yml

I'm unsure how this ever worked, but I think the inputs were largely being ignored. Please remove them now because they are breaking action runs

https://github.com/canonical/k8s-operator/actions/workflows/publish-charms.yaml

### CODE REVIEW

**Comment by addyess** [comment]
FYI ^^ @arturo-seijas @cbartz

**Review by arturo-seijas — APPROVED** [review]

**Comment by addyess** [comment]
@cbartz I hope you can look at this and hopefully merge soon.  We're staring at a charm release next week and i'm getting nervous

**Comment by cbartz** [comment]
/canonical/self-hosted-runners/run-workflows f253b983b2c533abe33e2308bfbb333703a72d3e

**Comment by cbartz** [comment]
@addyess I think the reason for your failed jobs is that branch protection is not enabled for the main branch: https://github.com/canonical/k8s-operator/actions/runs/8851049655/job/24306730069#step:2:20 The rest api call to `/repos/{owner}/{repo}/branches/{branch}/protection/required_status_checks` returns a 404. See also https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#get-status-checks-protection

Regarding your PR: https://github.com/octokit/request-action/tree/v2.3.0?tab=readme-ov-file#inputs indicates that the matching parameters `owner`, `repo`, `branch` must be passed as input to the action, so I believe removing them is wrong. 

Long story short, please make sure that the main branch of your repos is protected (with the required status checks).

On our end, we'll need to update the documentation for the `publish_charm` workflow.

**Comment by addyess** [comment]
Hey hey you're so right. Thanks for catching that 
https://github.com/octokit/request-action/issues/26

Makes that very clear

**Comment by addyess** [comment]
@cbartz i'm still confused

i have a branch protection rule for main

image


In it i have required status checks
image


The doc you linked https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#get-status-checks-protection

indicates you'll get a 404 if ...
> The token must have the following permission set:
>
> administration:read

Perhaps the `REPO_ACCESS_TOKEN` created doesn't have the admin permissions?

**Comment by cbartz** [comment]
@addyess This could be the reason. A classical PAT with full repo rights (also mentioned in the [README](https://github.com/canonical/operator-workflows/blob/main/README.md)) should suffice.
image

**Comment by addyess** [comment]
@cbartz 
That was it.  I had to use my gh account to create the PAT for the repo.  I'm glad i have admin rights on the repo.
